server/chat-plugins/smogtours.ts

```diff
@@ -0,0 +1,428 @@
+/**
+ * Integration for Smogon tournaments.
+ * @author mia-pi-git
+ */
+import {FS, Utils} from '../../lib';
+
+interface TourEvent {
+	title: string;
+	url: string;
+	desc: string;
+	image?: {url: string, dimensions: [number, number]};
+	id: string;
+	shortdesc: string;
+}
+
+interface TourTable {
+	title: string;
+	tours: TourEvent[];
+	whitelist?: string[];
+	icon?: [string, number, number];
+}
+
+export const tours: Record<string, TourTable> = {
+	official: {
+		title: "Smogon Officials",
+		// cap this one's dimensions
+		icon: ['https://www.smogon.com/media/zracknel-beta.svg.m.1', 178, 200],
+		tours: [],
+	},
+	smogon: {
+		title: "Open Sign-Ups",
+		tours: [],
+	},
+	ps: {
+		title: "Pokemon Showdown!",
+		icon: ['https://play.pokemonshowdown.com/pokemonshowdownbeta.png', 146, 44],
+		tours: [],
+	},
+};
+try {
+	const data = JSON.parse(FS('config/chat-plugins/smogtours.json').readSync());
+	// settings should prioritize hardcoded values for these keys
+	const PRIO = ['title', 'icon'];
+	for (const key in data) {
+		const section = (tours[key] ||= data[key]) as any;
+		for (const k in data[key]) {
+			if (PRIO.includes(k)) {
+				if (!section[k]) section[k] = data[key][k];
+			} else {
+				section[k] = data[key][k];
+			}
+		}
+	}
+} catch {}
+
+function saveTours() {
+	FS('config/chat-plugins/smogtours.json').writeUpdate(() => JSON.stringify(tours));
+}
+
+function getTour(category: string, id: string) {
+	category = toID(category);
+	id = toID(id);
+	return tours?.[category].tours.find(f => f.id === id) || null;
+}
```

Comment:
```suggestion
function getTour(categoryId: ID, tournamentName: string) {
	const tournamentId = toID(tournamentName);
	return tours?.[categoryId].tours.find(f => f.id === tournamentId) || null;
}
```
I think forcing the category ID out of the function ensures more type safety, but nonetheless I feel like more variable name clarity is useful